### PR TITLE
Uncheck hide development modules

### PR DESCRIPTION
**Pull request by lemon-suse**
Add schedule files to add module to uncheck hided development modules.

- Related ticket: https://progress.opensuse.org/issues/160700
- Related MR: https://gitlab.suse.de/qe-yam/openqa-job-groups/-/merge_requests/231
- Needles: N/A
- Verification run: https://openqa.suse.de/tests/overview?build=lemon-suse%2Fos-autoinst-distri-opensuse%23check-out-hide-development-module&version=15-SP6&distri=sle

### CODE REVIEW

File: lib/Installation/ModuleRegistration/ModuleRegistrationController.pm [hunk 1]
```diff
@@ -63,4 +63,9 @@ sub get_modules {
     return $self->get_module_registration_page()->get_modules();
 }
 
+sub uncheck_hide_development_versions {
```
Comment (jknphy) [inline lib/Installation/ModuleRegistration/ModuleRegistrationController.pm:66]: you already have that method above `view_development_versions`.

**Comment by jknphy** [comment]
what could be done is to create a file 15sp5 in lib/Distribution/Sle which inherit from 15_current where you would override the method to show a message "No development versions to view" and have in 15_current to always view and untick the checkbox, as SP6 for now have it and SP7 will have it. Once the modules will not be in dev version we remove the file but the test doesn't need to be touch.

**Comment by jknphy** [comment]
this will break:
https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=register_extensions_and_modules&module_re=&group_glob=&not_group_glob=&comment=&distri=sle&version=15-SP6&build=92.1&groupid=129#

**Comment by lemon-suse** [comment]
> what could be done is to have a controller where you will view the dev version and another when you will show a message "no dev version" (so the test never changes), I'm not sure about the exact solution, but involves changes in:
> 
> lib/Distribution/Sle lib/DistributionProvider.pm
> 
> You might need to clean-up a bit the distribution path that we are not using anymore.

Could we simply to add function to judge whether the checkbox exist or not, ex:

 $testapi::distri->get_module_registration()->uncheck_hide_development_versions() if $testapi::distri->get_module_registration()->uncheck_checkbox_exist();

**Comment by jknphy** [comment]
> > what could be done is to have a controller where you will view the dev version and another when you will show a message "no dev version" (so the test never changes), I'm not sure about the exact solution, but involves changes in:
> > lib/Distribution/Sle lib/DistributionProvider.pm
> > You might need to clean-up a bit the distribution path that we are not using anymore.
> 
> Could we simply to add function to judge whether the checkbox exist or not, ex:
> 
> $testapi::distri->get_module_registration()->uncheck_hide_development_versions() if $testapi::distri->get_module_registration()->uncheck_checkbox_exist();

no, because then you need a specific test to know that the checkbox has really disappear when it should. The framework with libyui-rest-api and the distribution pattern is based on this to avoid precisely conditions.

**Comment by jknphy** [comment]
I was checking again and I don't think it worth the time to implement the solution by code in this case, we have few test (1 or more) that are going to need to tick the checkbox (because we don't add WE everywhere and we also have textmode one that do not needed. This is the matrix to not break:
https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=register_extensions_and_modules&module_re=&group_glob=&not_group_glob=&comment=&distri=sle&version=&build=20240614-1&groupid=421#
https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=register_extensions_and_modules&module_re=&group_glob=&not_group_glob=&comment=&distri=sle&version=15-SP6&build=93.5&groupid=129#

So, instead of changing the code and affect all the test suite above, you can change create new schedules for this case that will include the existing test module `installation/module_registration/view_development_versions.pm` and in some point of time we will remove them because the checkbox will disappear and we will put the same than for SP5 (but we still have full control of what is going on in the screen, otherwise, automation that acts depending of what it finds in the screen is hard to maintain).

Please, try to create those new schedules, should be pretty straighforward.

**Comment by lemon-suse** [comment]
> I was checking again and I don't think it worth the time to implement the solution by code in this case, we have few test (1 or more) that are going to need to tick the checkbox (because we don't add WE everywhere and we also have textmode one that do not needed. This is the matrix to not break: https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=register_extensions_and_modules&module_re=&group_glob=&not_group_glob=&comment=&distri=sle&version=&build=20240614-1&groupid=421# https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=register_extensions_and_modules&module_re=&group_glob=&not_group_glob=&comment=&distri=sle&version=15-SP6&build=93.5&groupid=129#
> 
> So, instead of changing the code and affect all the test suite above, you can change create new schedules for this case that will include the existing test module `installation/module_registration/view_development_versions.pm` and in some point of time we will remove them because the checkbox will disappear and we will put the same than for SP5 (but we still have full control of what is going on in the screen, otherwise, automation that acts depending of what it finds in the screen is hard to maintain).
> 
> Please, try to create those new schedules, should be pretty straightforward.

Ok, this is not the best method but can solve this issue for now:). Even we change code to judge the version, still need to update the condition of released version later. 
Updated the code and add related MR, VR updated.

File: lib/Installation/ModuleRegistration/ModuleRegistrationController.pm (hunk 1 above)
Comment (lemon-suse) [reply lib/Installation/ModuleRegistration/ModuleRegistrationController.pm]: Ok, use the existing method.

File: schedule/yast/maintenance/lvm_thin_provisioning_view_development.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+name: lvm_thin_provisioning_view_development
+vars:
+  DUD_ADDONS: sdk
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/view_development_versions.pm
```
Comment (jknphy) [inline schedule/yast/maintenance/lvm_thin_provisioning_view_development.yaml:8]: you don't need .pm.

File: tests/installation/module_registration/view_development_versions.pm [hunk 3]
```diff
@@ -1,16 +1,17 @@
 # SUSE's openQA tests
 #
-# Copyright 2021 SUSE LLC
+# Copyright 2022 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 
-# Summary: View list of available extension and modules including
-# development versions in module registration.
+# Summary: Uncheck 'hide development versions'
```
Comment (jknphy) [inline tests/installation/module_registration/view_development_versions.pm:6]: the current description was more readable. "View" is more general, 'check', 'tick' 'untick' all these stuff is better for another layer.

File: tests/installation/module_registration/view_development_versions.pm [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
+#          in "Extension and Module Selection" dialog.
 #
 # Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
 
 use base 'y2_installbase';
 use strict;
 use warnings;
+use testapi qw(save_screenshot get_required_var);
```
Comment (jknphy) [inline tests/installation/module_registration/view_development_versions.pm:14]: you don't need to add this line or change this file at all.

**Comment by jknphy** [comment]
ahm I thought that we only use WE for few test suite, but we use it for all, then why don't we create a new default yaml file that contains that step? It will save files, right?

**Comment by lemon-suse** [comment]
> ahm I thought that we only use WE for few test suite, but we use it for all, then why don't we create a new default yaml file that contains that step? It will save files, right?

Yes, add the default schedule and updated MR.

File: schedule/yast/qam-yast_self_update_view_development.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (jknphy) [inline schedule/yast/qam-yast_self_update_view_development.yaml:1]: could we quickly convert this one to yaml based on defaut with keys?

**Pull request merged**